Doc/whatsnew/3.12.rst

```diff
@@ -102,6 +102,29 @@ Deprecated
 Removed
 =======
 
+* Removed many old deprecated :mod:`unittest` features:
+
+  - :class:`~unittest.TestCase` method aliases ``failUnlessEqual``,
```

Comment:
Do we have tooling that can updated people's test code for them?  I think this change will go over better if we can provide a tool.  I suggest adding a fixer to https://github.com/PyCQA/modernize that updates old `TestCase` method names so we can link to that and provide a specific command line for people just run that fixer on their `_*test.py` files.

Reply:
I did not use it, but people recommended https://github.com/isidentical/teyit for it.

Reply:
@isidentical what do you think about us suggesting your teyit tool in the what's new entry as a way for people to automate fixing up their old test code to use the canonical unittest TestCase assertion names as we remove the ancient deprecated ones?

Reply:
I've used teyit specifically for preparing for these removals, and can recommend it. 👍

There may be a couple of less-common methods not covered, it would be good to review and add them if possible.

Reply:
Suggesting teyit LGTM.